api/base/controllers/roles.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+var simpleDI = require('config/simpleDI');
+
+module.exports = simpleDI.inject(['mongoose', 'base/roleModel', 'jsonwebtoken', 'app/config'], function (mongoose, Role, jwt, appConfig) {
+
+  var ObjectId = mongoose.Types.ObjectId;
+
+  return {
+
+    /**
+     * Create role
+     * requires: {roleName}
+     * returns: {message}
+     */
+      create: function (req, res, next) {
+          var newRole = new Role();
+          newRole.roleName = req.body.roleName;
+          try {
+            newRole.save(function(err) {
+              if (err) {
+                return res.json(400, { message: err });
+              }
+                          
+              res.json(200, { message: 'Role created.' });
+            });
+          }
+          catch (err){
+              return res.json(400, { message: err });            
+          }
+        },
+    
+      /**
+       *  get all roles
+       *  returns {_id, roleName}
+       */
+      getAll: function (req, res, next) {          
+        
+          Role.find({}, function (err, roles) {
+            if (err) {
+              return next(new Error('Failed to load roles'));
+            }
+            if (roles) {
+              res.json(roles);
+            } else {
+              res.json(404, { message: 'Roles not found' });
+            }
+          });
+      },
+      
+      getById: function (req, res, next) {
+         var roleId = req.params.roleId;
+        
+          Role.findById(ObjectId(roleId)).exec(function (err, role) {
+            if (err) {

```

Comment:
I would send the error from the database and do something like: `if (err) { return next(err); }`